src/app/screens/confirmStxTransaction/index.tsx

```diff
@@ -123,7 +134,7 @@ function ConfirmStxTransaction() {
     }
 
     const txAmount = new BigNumber(txPayload.amount.toString(10));
```

Comment:
Instead of having updateUI as a method that's called from the useEffect, could you rather put it directly inside the useEffect?

Also, the useEffect will always call updateUI because on load those states will always be empty. It might be better to try and parse the values and set the initial states directly, otherwise remove the `if` in the useEffect and add an empty dependency array as the second arg.